akka-sample-spring-extension-java/src/main/java/org/example/akka/di/SpringExtension.java

```diff
@@ -0,0 +1,24 @@
+package org.example.akka.di;
+
+import akka.actor.Extension;
+import akka.actor.Props;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpringExtension implements Extension {
+
+	private ApplicationContext applicationContext;
```

Comment:
Extensions are responsible to be thread safe, given that this field must be `volatile` or an `AtomicReference` else calls to the `props` methods (from different threads) may not see that it was initialized with any value

Reply:
Is making the Props methods Thread-safe with the "synchronized" be enough?